### ISSUE TITLE
Renaming properties using custom contract resolver has no effect on the swagger document

### ISSUE THREAD

**Issue by statler** [RSuter/NSwag#1270]
Renaming properties per the blog examples does affect the swagger document. The property is properly renamed in the CreateProperty method, but the generated swagger JSON is unchanged. Do I misunderstand how this works?

        protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
        {
            var property = base.CreateProperty(member, memberSerialization);
            var className = property.DeclaringType.Name;

            if (IsIncluded(property.DeclaringType, property.PropertyName)) property.ShouldSerialize = i => true;
            else if (_globalIgnores.Contains(property.PropertyName) || IsIgnored(property.DeclaringType, property.PropertyName))
                property.ShouldSerialize = i => false;
            else if (_TypeIgnoreNonPrimitive.Contains(property.DeclaringType.Name))
            {
                if (!property.PropertyType.IsPrimitive && !(property.PropertyType==typeof(string)))
                    property.ShouldSerialize = i => false;
            }
            else if (hideComplexTypeCollections && property.PropertyType.IsGenericType && (property.PropertyType.GetGenericTypeDefinition() == typeof(ICollection<>) || property.PropertyType.GetGenericTypeDefinition() == typeof(IEnumerable<>)))
                property.ShouldSerialize = i => false;
            else property.ShouldSerialize = i => true;

            if (IsRenamed(property.DeclaringType, property.PropertyName, out var newJsonPropertyName))
                property.PropertyName = newJsonPropertyName;

            return property;
        }

**Comment by RSuter** [comment]
Did you register the custom contract resolver in the generator settings?

**Comment by statler** [comment]
Yes. I have ignores working perfectly, and as per the original issue, it works if I set both PropertyName and UnderlyingName

**Comment by RSuter** [comment]
Strange that you also have to rename the UnderlyingName... any idea why? If this is the case, I should probably update the blog...

**Comment by statler** [comment]
Don't know why - not that it really matters when you think about it, because renaming an item in the contract is inherently useless if you can't also include the original name in the swagger json. In the end, I have done the renaming in the client (refer https://github.com/statler/NJsonSchema)

**Comment by tmacht** [comment]
> Did you register the custom contract resolver in the generator settings?

How do you register a custom contract resolver in the generator settings?

**Comment by RicoSuter** [comment]
Register it in the mvc json options
